app/models/manageiq/providers/ansible_runner_workflow.rb

```diff
@@ -22,11 +22,26 @@ def execution_type
     raise NotImplementedError, "must be implemented in a subclass"
   end
 
+  def start
+    queue_signal(:pre_execute)
+  end
+
   def pre_execute
-    # A step before running the playbook/role for any optional setup tasks
+    prepare_repository
     queue_signal(:execute)
   end
 
+  def adjust_options_for_git_checkout_dir!
+    raise NotImplementedError, "must be implemented in a subclass"
+  end
+
+  def prepare_repository
+    return unless options[:configuration_script_source_id]
```

Comment:
I feel like we should have some verification here that we have both the source id and the playbook or role relative path. We know we would fail later on if we don't have both.

Reply:
Do you suggest raising an exception if we get one or the other but not both?